lib/ReactViews/DataCatalogGroup.jsx

```diff
@@ -1,64 +1,49 @@
     mixins: [ObserveModelMixin],
 
     propTypes: {
-        group: React.PropTypes.object,
-        previewedCatalogItem: React.PropTypes.object,
-        onPreviewedCatalogItemChanged: React.PropTypes.func,
-        isOpen: React.PropTypes.bool,
-        onToggleOpen: React.PropTypes.func
+        group: React.PropTypes.object.isRequired,
+        viewState: React.PropTypes.object.isRequired
     },
 
-    toggleOpen(e) {
-        if (defined(this.props.onToggleOpen)) {
-            this.props.onToggleOpen();
-        } else {
-            this.props.group.toggleOpen();
-        }
-    },
-
-    renderGroup(group, isOpen) {
-        if (isOpen === true) {
-            if (group.items && group.items.length > 0) {
-                return group.items.map((member, i)=>{
-                    if (member.isGroup) {
-                        return (<DataCatalogGroup group={member}
-                                                  key={i}
-                                                  previewedCatalogItem={this.props.previewedCatalogItem}
-                                                  onPreviewedCatalogItemChanged={this.props.onPreviewedCatalogItemChanged}
-                                 />);
-                    }
-                    return (<DataCatalogItem item={member}
-                                             key={i}
-                                             previewedCatalogItem={this.props.previewedCatalogItem}
-                                             onPreviewedCatalogItemChanged={this.props.onPreviewedCatalogItemChanged}
-                            />);
-                });
-            }
-            return <Loader/>;
-        }
+    toggleOpen() {
+        this.props.group.isOpen = !this.props.group.isOpen;
     },
 
     render() {
         const group = this.props.group;
-        const isOpen = defined(this.props.isOpen) ? this.props.isOpen : group.isOpen;
 
         return (
             <li>
-              <button className ={'btn btn-catalogue ' + (isOpen ? 'is-open' : '')} onClick={this.toggleOpen} >{group.name} <i className={'icon ' + (isOpen ? 'icon-caret-down' : 'icon-caret-right')}></i></button>
-              <ul className="data-catalog-group list-reset">
-                {this.renderGroup(group, isOpen)}
-              </ul>
+                <button className={classNames('btn', 'btn-catalogue', {'is-open' : group.isOpen})}
+                        onClick={this.toggleOpen}>
+                    {group.name}
+                    <i className={classNames('icon', {'icon-caret-down': group.isOpen}, {'icon-caret-right': !group.isOpen})} />
+                </button>
+                {group.isOpen && (
+                    <ul className="data-catalog-group list-reset">
+                        {this.renderGroup(group)}
+                    </ul>
+                )}
             </li>
-            );
+        );
+    },
+
+    renderGroup(group) {
+        if (!group.isLoading) {
```

Comment:
We want to incrementally show the items even while we're loading.  This is important for the CSW group, for example, because it takes _forever_ to load everything, but the first 100 items appear fairly quickly.